### PR TITLE
Enable strict explicit api mode and fix errors

### LINKED ISSUE #25: Introduce explicit-api mode

### PR DESCRIPTION
**Pull request by IgnatBeresnev**
Fixes #25

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -17,8 +17,11 @@ repositories {
 dependencies {
     implementation("com.google.protobuf:protobuf-java:2.6.1")
     implementation("org.jetbrains.kotlin:kotlin-stdlib")
+
+    val kotlinx_metadata_version: String by project
+    implementation("org.jetbrains.kotlinx:kotlinx-metadata-jvm:$kotlinx_metadata_version")
```
Comment (IgnatBeresnev) [inline build.gradle.kts:22]: Not related to explicit api mode, just noticed that metadata-jvm version was specified in two places (in here and an unused property in `gradle.properties`)

File: build.gradle.kts [hunk 2]
```diff
@@ -3,7 +3,7 @@
  */
 
 plugins {
-    kotlin("jvm") version "1.7.10"
+    kotlin("jvm")
```
Comment (IgnatBeresnev) [inline build.gradle.kts:6]: Not related to explicit api mode, was passing by and noticed a potentially confusing duplication of used Kotlin version

Resolution happens in [settings.gradle](https://github.com/Kotlin/kotlinx.reflect.lite/blob/23297f144bc2eda7d2ca6ae0aeadaa17c23f29e4/settings.gradle.kts#L8-L15) based on `kotlin_version` gradle property

File: src/main/java/kotlinx/reflect/lite/impl/KotlinReflectionInternalError.kt [hunk 3]
```diff
@@ -9,4 +9,4 @@ package kotlinx.reflect.lite.impl
  * Signals that Kotlin reflection had reached an inconsistent state from which it cannot recover.
  * @suppress
  */
-class KotlinReflectionInternalError(message: String) : Error(message)
+public class KotlinReflectionInternalError(message: String) : Error(message)
```
Comment (IgnatBeresnev) [inline src/main/java/kotlinx/reflect/lite/impl/KotlinReflectionInternalError.kt:12]: Not sure about this one, but I guess errors that the user can end up seeing should be public, even if there's nothing sensible they can do about it?

File: build.gradle.kts (hunk 2 above)
Comment (qwwdfsad) [reply build.gradle.kts:6]: Good catch

File: src/main/java/kotlinx/reflect/lite/impl/KotlinReflectionInternalError.kt (hunk 3 above)
Comment (qwwdfsad) [reply src/main/java/kotlinx/reflect/lite/impl/KotlinReflectionInternalError.kt:12]: The honest question is "we don't know yet".

>I guess errors that the user can end up seeing should be public

Not necessary, as long as it has a meaningful (for the API) superclass/superinterface.
This one is weird -- it takes its original place in `kotlin.reflect.jvm.internal` package and probably was made `public` by mistake, but people are [already using it](https://grep.app/search?current=2&q=KotlinReflectionInternalError&filter[lang][0]=Kotlin)

**Review by qwwdfsad — APPROVED** [review]
Thanks for taking care of it! 👍

**Comment by qwwdfsad** [comment]
I've pushed a commit that migrates from `-X` flag to official DSL that otherwise introduces no semantics changes

**Pull request merged**